<a href="https://colab.research.google.com/github/manudsanun/python_dataviz2024/blob/main/Melanoma.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
from google.colab import drive

In [ ]:
drive.mount('/content/drive')

Mounted at /content/drive


# นำเข้าข้อมูลยีนจาก NCBI


1. "matches ensembl"[Properties]
เป็นการค้นหาข้อมูลใน NCBI ที่มีการจับคู่ (matches) หรือสอดคล้องกับข้อมูลจาก Ensembl database.
ใช้เพื่อกรองเฉพาะรายการที่มีการอ้างอิงหรือเชื่อมโยงกับ transcript, gene, หรือ protein ID จาก Ensembl.

2. alive[prop]
เป็นการกรองข้อมูลเพื่อแสดงเฉพาะรายการที่ยัง active หรือมีสถานะใช้งาน (alive) ในระบบฐานข้อมูล.
ช่วยตัดข้อมูลที่ถูกระงับหรือเลิกใช้งานออกไป.

In [ ]:
# melanoma[All Fields] AND ("matches ensembl"[Properties] AND alive[prop])
ncbi = pd.read_csv('/content/drive/MyDrive/Spatial/NCBI_melanoma.txt', delimiter='\t')
ncbi

tax_id            Org_name     GeneID  CurrentID Status   Symbol  \
0       9606        Homo sapiens       1029          0   live   CDKN2A   
1       9606        Homo sapiens      51151          0   live  SLC45A2   
2       9606        Homo sapiens       1026          0   live   CDKN1A   
3       9606        Homo sapiens      64135          0   live    IFIH1   
4       9606        Homo sapiens       4157          0   live     MC1R   
...      ...                 ...        ...        ...    ...      ...   
7497    8355      Xenopus laevis  108706844          0   live  stk11.S   
7498    8355      Xenopus laevis  108706736          0   live  nfkb1.S   
7499    8355      Xenopus laevis  108700512          0   live  ass1p.S   
7500    8355      Xenopus laevis  108699122          0   live   naf1.L   
7501    8364  Xenopus tropicalis  100380119          0   live  ccdc71l   

                                                Aliases  \
0     ARF, CAI2, CDK4I, CDKN2, CMM2, INK4, INK4A, ML...   
1                          1A1, AIM1, MATP, OCA4, SHEP5   
2     CAP20, CDKN1, CIP1, MDA-6, P21, SDI1, WAF1, p2...   
3     AGS7, Hlcd, IDDM19, IMD95, MDA-5, MDA5, RLR-2,...   
4                                    CMM5, MSH-R, SHEP2   
...                                                 ...   
7497                     XELAEV_18009773mg, LKB1, XEEK1   
7498  XELAEV_18009414mg, ebp-1, kbf1, nf-kappa-b, nf...   
7499                                                ass   
7500                                  XELAEV_18005436mg   
7501                                  c7orf74, flj36031   

                                            description  \
0                  cyclin dependent kinase inhibitor 2A   
1                     solute carrier family 45 member 2   
2                  cyclin dependent kinase inhibitor 1A   
3           interferon induced with helicase C domain 1   
4                               melanocortin 1 receptor   
...                                                 ...   
7497              serine/threonine kinase 11 S homeolog   
7498  nuclear factor of kappa light polypeptide gene...   
7499  argininosuccinate synthase 1 pseudogene S home...   
7500  nuclear assembly factor 1 ribonucleoprotein L ...   
7501              coiled-coil domain containing 71 like   

                                     other_designations map_location  \
0     cyclin-dependent kinase inhibitor 2A|CDK4 inhi...       9p21.3   
1     membrane-associated transporter protein|antige...       5p13.2   
2     cyclin-dependent kinase inhibitor 1|CDK-intera...       6p21.2   
3     interferon-induced helicase C domain-containin...       2q24.2   
4     melanocyte-stimulating hormone receptor|MC1-R|...      16q24.3   
...                                                 ...          ...   
7497         serine/threonine-protein kinase stk11-like          NaN   
7498             nuclear factor NF-kappa-B p105 subunit          NaN   
7499  argininosuccinate synthase 1 S homeolog|argini...          NaN   
7500  H/ACA ribonucleoprotein complex non-core subun...          NaN   
7501          coiled-coil domain-containing protein 71L          NaN   

     chromosome genomic_nucleotide_accession.version  \
0             9                         NC_000009.12   
1             5                         NC_000005.10   
2             6                         NC_000006.12   
3             2                         NC_000002.12   
4            16                         NC_000016.10   
...         ...                                  ...   
7497         1S                          NC_054372.1   
7498         1S                          NC_054372.1   
7499         8S                          NC_054386.1   
7500         1L                          NC_054371.1   
7501          3                          NC_030679.2   

      start_position_on_the_genomic_accession  \
0                                  21967752.0   
1                                  33944623.0   
2             

NCBI => Gene Symbol มีหลักการตั้งชื่อเป็นไปตามมาตรฐานของ HGNC (HUGO Gene Nomenclature Committee) ในมนุษย์

ดังนั้น ในตารางข้อมูล Melanoma เราต้อง filter เอาเฉพาะ rows ที่มีการตั้งชื่อแบบ HGNC

# นำเข้าข้อมูล Immuno-oncology Gene Panel

In [ ]:
Mela = pd.read_csv('/Immuno-oncology Gene Panel _Gene List.csv', encoding='latin-1')
# or 'cp1252' or any other encoding that might work for your data
Mela.head()

Gene                                             Classes  \
0    ACKR3  Atypical chemokine receptors;Cytokine Receptor...   
1    ACTA2                      Cancer-associated fibroblasts   
2  ADAMTS4                                  Matrix remodeling   
3     AKT1                                   Cancer Signaling   
4     AKT2                                   Cancer Signaling   

                                         Description               ID  \
0                      atypical chemokine receptor 3  ENST00000272928   
1  Myofibroblasts acquire contractile stress fibe...  ENST00000224784   
2  ADAMTSs are a family of secreted proteinases t...  ENST00000367996   
3                      AKT serine/threonine kinase 1  ENST00000407796   
4                      AKT serine/threonine kinase 2  ENST00000392038   

      HGNC ID Location  Unnamed: 6  Unnamed: 7  
0  HGNC:23692   2q37.3         NaN         NaN  
1         NaN      NaN         NaN         NaN  
2         NaN      NaN         NaN         NaN  
3    HGNC:391      NaN         NaN         NaN  
4    HGNC:392  19q13.2         NaN         NaN

In [ ]:
Mela_HGNC = Mela.dropna(subset=['HGNC ID']) # Specify the column using the 'subset' parameter
Mela_HGNC

Gene                                             Classes  \
0     ACKR3  Atypical chemokine receptors;Cytokine Receptor...   
3      AKT1                                   Cancer Signaling   
4      AKT2                                   Cancer Signaling   
5      AKT3                                   Cancer Signaling   
6    AMOTL2                     Beta-catenin and cell skeleton   
..      ...                                                ...   
492   WWTR1                                      Hippo pathway   
493    XBP1                         Transcriptional Regulation   
496    YAP1                                      Hippo pathway   
498   ZBED2   Transcriptional regulator and competes iwht IRF1   
499    ZEB1       Transcriptional repressor in immune response   

                                        Description               ID  \
0                     atypical chemokine receptor 3  ENST00000272928   
3                     AKT serine/threonine kinase 1  ENST00000407796   
4                     AKT serine/threonine kinase 2  ENST00000392038   
5                     AKT serine/threonine kinase 3  ENST00000263826   
6                                 angiomotin like 2  ENST00000422605   
..                                              ...              ...   
492  WW domain containing transcription regulator 1  ENST00000465804   
493                         X-box binding protein 1  ENST00000216037   
496       Yes1 associated transcriptional regulator  ENST00000282441   
498               zinc finger BED-type containing 2  ENST00000317012   
499            zinc finger E-box binding homeobox 1  ENST00000320985   

        HGNC ID  Location  Unnamed: 6  Unnamed: 7  
0    HGNC:23692    2q37.3         NaN         NaN  
3      HGNC:391       NaN         NaN         NaN  
4      HGNC:392   19q13.2         NaN         NaN  
5      HGNC:393  1q43-q44         NaN         NaN  
6    HGNC:17812    3q22.2         NaN         NaN  
..          ...       ...         ...         ...  
492  HGNC:24042    3q25.1         NaN         NaN  
493  HGNC:12801       NaN         NaN         NaN  
496  HGNC:16262   11q22.1         NaN         NaN  
498  HGNC:20710   3q13.13         NaN         NaN  
499  HGNC:11642  10p11.22         NaN         NaN  

[249 rows x 8 columns]

# หายีนที่ match กันในระหว่างคอลัมน์ "Gene" และ "Symbol"

In [ ]:
# แสดงเฉพาะคอลัมน์ Symbol
ncbi1 = ncbi['Symbol']

# แสดงเฉพาะคอลัมน์ Gene
Mela_HGNC1 = Mela_HGNC['Gene ']

# เปลี่ยนชื่อคอลัมน์ให้เหมือนกันเพื่อความสะดวกในการเปรียบเทียบ
Mela_HGNC1 = Mela_HGNC1.rename('Symbol')

# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
merged_df = pd.merge(ncbi1, Mela_HGNC1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes_count = len(merged_df)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่ตรงกันระหว่างตาราง Mela_HGNC และ ncbi คือ: {matched_genes_count} ตัว")

# แสดงตารางที่รวมกันแล้ว (ถ้าต้องการ)
merged_df

จำนวนยีนที่ตรงกันระหว่างตาราง Mela_HGNC และ ncbi คือ: 134 ตัว


Symbol
0    CDKN1A
1     STAT3
2     NFKB1
3      BRAF
4       MYC
..      ...
129  TGFBR1
130    FLT4
131    MUC4
132   PDGFA
133    JUNB

[134 rows x 1 columns]

In [ ]:
# prompt: นำข้อมูลจาก merged_df ออกเป็นไฟล์ excel

# Specify the output file path in your Google Drive, including the file extension (.xlsx)
output_file_path = '/content/drive/MyDrive/Spatial/output_file.xlsx'  # Changed this to include .xlsx

# Export the merged DataFrame to an Excel file
merged_df.to_excel(output_file_path, index=False)  # index=False prevents writing row indices to the file

print(f"Merged data exported to: {output_file_path}")

Merged data exported to: /content/drive/MyDrive/Spatial/output_file.xlsx


จริงๆใน NCBI มีหลักการตั้งชื่อแบบ HGNC แต่ Gene name เป็นชื่อเดียวกับตั้งแบบ ENST ดังนั้นเราจะทำการ Match ข้อมูลจาก panel ทั้ง 500 gene กับ ข้อมูลจาก NCBI เลย


In [ ]:
# แสดงเฉพาะคอลัมน์ Symbol
ncbi = ncbi['Symbol']

# แสดงเฉพาะคอลัมน์ Gene
Mela1 = Mela['Gene ']

# เปลี่ยนชื่อคอลัมน์ให้เหมือนกันเพื่อความสะดวกในการเปรียบเทียบ
Mela1 = Mela1.rename('Symbol')

# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
merged_df = pd.merge(ncbi, Mela1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes_count = len(merged_df)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่ตรงกันระหว่างตาราง Mela และ ncbi คือ: {matched_genes_count} ตัว")

# แสดงตารางที่รวมกันแล้ว (ถ้าต้องการ)
merged_df

จำนวนยีนที่ตรงกันระหว่างตาราง Mela และ ncbi คือ: 269 ตัว


Symbol
0       IFNG
1      FOXP3
2     CDKN1A
3      CXCL1
4      IL12A
..       ...
264     MUC4
265    PDGFA
266     JUNB
267     CD68
268  SHARPIN

[269 rows x 1 columns]

In [ ]:
# ncbi_new : immune[All Fields] AND melanoma[All Fields]
ncbi_new = pd.read_csv('/content/drive/MyDrive/Spatial/gene_ncbi(1).txt', delimiter='\t')
ncbi_new.head()

tax_id      Org_name  GeneID  CurrentID Status  Symbol  \
0    9606  Homo sapiens    1029          0   live  CDKN2A   
1    9606  Homo sapiens    3458          0   live    IFNG   
2    9606  Homo sapiens   50943          0   live   FOXP3   
3    9606  Homo sapiens   64135          0   live   IFIH1   
4    9606  Homo sapiens    9447          0   live    AIM2   

                                             Aliases  \
0  ARF, CAI2, CDK4I, CDKN2, CMM2, INK4, INK4A, ML...   
1                                    IFG, IFI, IMD69   
2                AIID, DIETER, IPEX, JM2, PIDX, XPID   
3  AGS7, Hlcd, IDDM19, IMD95, MDA-5, MDA5, RLR-2,...   
4                                             PYHIN4   

                                   description  \
0         cyclin dependent kinase inhibitor 2A   
1                             interferon gamma   
2                              forkhead box P3   
3  interferon induced with helicase C domain 1   
4                         absent in melanoma 2   

                                  other_designations  map_location chromosome  \
0  cyclin-dependent kinase inhibitor 2A|CDK4 inhi...        9p21.3          9   
1       interferon gamma|IFN-gamma|immune interferon         12q15         12   
2  forkhead box protein P3|FOXP3delta7|immune dys...       Xp11.23          X   
3  interferon-induced helicase C domain-containin...        2q24.2          2   
4                  interferon-inducible protein AIM2  1q23.1-q23.2          1   

  genomic_nucleotide_accession.version  \
0                         NC_000009.12   
1                         NC_000012.12   
2                         NC_000023.11   
3                         NC_000002.12   
4                         NC_000001.11   

   start_position_on_the_genomic_accession  \
0                               21967752.0   
1                               68154768.0   
2                               49250438.0   
3                              162267074.0   
4                              159055051.0   

   end_position_on_the_genomic_accession orientation  exon_count    OMIM  \
0                             21995324.0       minus        10.0  600160   
1                             68159740.0       minus         4.0  147570   
2                             49264710.0       minus        12.0  300292   
3                            162318684.0       minus        17.0  606951   
4                            159147132.0       minus        14.0  604578   

   Unnamed: 17  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN

In [ ]:
# แสดงเฉพาะคอลัมน์ Gene
Mela1 = Mela['Gene ']

# เปลี่ยนชื่อคอลัมน์ให้เหมือนกันเพื่อความสะดวกในการเปรียบเทียบ
Mela1 = Mela1.rename('Symbol')

# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
merged_df1 = pd.merge(ncbi_new, Mela1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes_count = len(merged_df1)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่ตรงกันระหว่างตาราง Mela และ ncbi_new คือ: {matched_genes_count} ตัว")

# แสดงตารางที่รวมกันแล้ว (ถ้าต้องการ)
merged_df1

จำนวนยีนที่ตรงกันระหว่างตาราง Mela และ ncbi_new คือ: 269 ตัว


Symbol
0       IFNG
1      FOXP3
2     CDKN1A
3      CXCL1
4      IL12A
..       ...
264     MUC4
265    PDGFA
266     JUNB
267     CD68
268  SHARPIN

[269 rows x 1 columns]

In [ ]:
# prompt: นำข้อมูลจาก merged_df1 ออกเป็นไฟล์ excel

# Specify the output file path in your Google Drive, including the file extension (.xlsx)
output_file_path = '/content/drive/MyDrive/Spatial/output_file.xlsx'  # Changed this to include .xlsx

# Export the merged DataFrame to an Excel file
merged_df1.to_excel(output_file_path, index=False)  # index=False prevents writing row indices to the file

print(f"Merged data exported to: {output_file_path}")

Merged data exported to: /content/drive/MyDrive/Spatial/output_file.xlsx


# เทียบกับ GeneCard

In [ ]:
GeneCard = pd.read_csv('/content/drive/MyDrive/Spatial/GeneCards-SearchResults.csv')
GeneCard

Gene Symbol                                        Description  \
0            CDKN2A               Cyclin Dependent Kinase Inhibitor 2A   
1              CDK4                          Cyclin Dependent Kinase 4   
2              MC1R                            Melanocortin 1 Receptor   
3              MITF           Melanocyte Inducing Transcription Factor   
4              BRAF      B-Raf Proto-Oncogene, Serine/Threonine Kinase   
...             ...                                                ...   
11068  LOC111089942                              Pleiotrophin Promoter   
11069  LOC117038771  CRISPRi-FlowFISH-Validated H1-10 Regulatory El...   
11070  LOC129937550  ATAC-STARR-Seq Lymphoblastoid Active Region 20512   
11071        ORI343                          Origin Of Replication 343   
11072         SSXP8                            SSX Family Pseudogene 8   

                 Category Uniprot ID  Gifts        GC Id  Relevance score  
0          Protein Coding     Q8N726     63  GC09M021967       146.845993  
1          Protein Coding     P11802     66  GC12M059932       116.223862  
2          Protein Coding     Q01726     59  GC16P089912        91.914803  
3          Protein Coding     O75030     61  GC03P069740        90.593117  
4          Protein Coding     P15056     66  GC07M140783        87.288742  
...                   ...        ...    ...          ...              ...  
11068  Functional Element        NaN      2  GC07P137343         0.101479  
11069  Functional Element        NaN      2  GC03P128487         0.101479  
11070  Functional Element        NaN      2  GC03P141520         0.101479  
11071  Functional Element        NaN      2  GC06P156840         0.101479  
11072          Pseudogene        NaN      2  GC0XP048276         0.101479  

[11073 rows x 7 columns]

ใน GeneCard มีข้อมูลทั้งหมด 11073 ยีน

In [ ]:
ncbi_new

0        CDKN2A
1          IFNG
2         FOXP3
3         IFIH1
4          AIM2
         ...   
1952    PLEKHA4
1953       RTP4
1954       SCFV
1955      TEX41
1956      nfkb1
Name: Symbol, Length: 1957, dtype: object

ใน NCBI มีข้อมูลทั้งหมด 1957 ยีน

เปรียบเทียบข้อมูลจาก NCBI กับ GeneCard

In [ ]:
# แสดงเฉพาะคอลัมน์ Gene
GeneCard1 = GeneCard['Gene Symbol']

# เปลี่ยนชื่อคอลัมน์ให้เหมือนกันเพื่อความสะดวกในการเปรียบเทียบ
GeneCard1 = GeneCard1.rename('Symbol')

# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
merged_df3 = pd.merge(ncbi_new, GeneCard1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes_count = len(merged_df3)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่ตรงกันระหว่างตาราง GeneCard และ NCBI คือ {matched_genes_count} ตัว")

# แสดงตารางที่รวมกันแล้ว (ถ้าต้องการ)
merged_df3

จำนวนยีนที่ตรงกันระหว่างตาราง GeneCard และ NCBI คือ 1451 ตัว


Symbol
0      CDKN2A
1        IFNG
2       FOXP3
3       IFIH1
4        AIM2
...       ...
1446  SPANXB1
1447  PLEKHA4
1448     RTP4
1449     SCFV
1450    TEX41

[1451 rows x 1 columns]

จำนวนยีนที่ตรงกันระหว่างข้อมูลจาก GeneCard1 และ NCBI คือ 1451 ยีน

เอาข้อมูล merged_df3 มา match กับ 500 gene (Mela)

In [ ]:
# แสดงเฉพาะคอลัมน์ Gene
Mela1 = Mela['Gene ']

# เปลี่ยนชื่อคอลัมน์ให้เหมือนกันเพื่อความสะดวกในการเปรียบเทียบ
Mela1 = Mela1.rename('Symbol')

# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
merged_melanoma = pd.merge(merged_df3, Mela1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes = len(merged_melanoma)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่ตรงกันระหว่างตาราง Mela1 และ merged_df3 คือ {matched_genes} ตัว")

# แสดงตารางที่รวมกันแล้ว (ถ้าต้องการ)
merged_melanoma

จำนวนยีนที่ตรงกันระหว่างตาราง Mela1 และ merged_df3 คือ 269 ตัว


Symbol
0       IFNG
1      FOXP3
2     CDKN1A
3      CXCL1
4      IL12A
..       ...
264     MUC4
265    PDGFA
266     JUNB
267     CD68
268  SHARPIN

[269 rows x 1 columns]

จำนวนยีนที่ตรงกันระหว่างตาราง Mela1 และ merged_df3 คือ 316 ตัว

In [ ]:
# prompt: นำข้อมูลออกเป็นไฟล์ excel

# Specify the output file path in your Google Drive, including the file extension (.xlsx)
output_merged_melanoma = '/content/drive/MyDrive/Spatial/output_file.xlsx'  # Changed this to include .xlsx

# Export the merged DataFrame to an Excel file
merged_melanoma.to_excel(output_merged_melanoma, index=False)  # index=False prevents writing row indices to the file

print(f"Merged data exported to: {output_merged_melanoma}")

Merged data exported to: /content/drive/MyDrive/Spatial/output_file.xlsx


# ใช้ ENST ID

In [ ]:
ENST = pd.read_csv('/content/drive/MyDrive/Spatial/ensembl_melanoma_human.csv')
ENST.head()

id_with_url              name species feature_type  \
0  ENST00000446972  novel transcript   Human   Transcript   
1  ENST00000412173  novel transcript   Human   Transcript   
2  ENST00000423985  novel transcript   Human   Transcript   
3  ENST00000439389  novel transcript   Human   Transcript   
4  ENST00000446214  novel transcript   Human   Transcript   

                                         description  
0                        melanoma antigen pseudogene  
1  MAGE (Melanoma Associated Antigen) family pseu...  
2  MAGE (Melanoma Associated Antigen) family pseu...  
3                        melanoma antigen pseudogene  
4  pseudogene similar to part of melanoma antigen...

In [ ]:
Mela

Gene                                             Classes  \
0      ACKR3  Atypical chemokine receptors;Cytokine Receptor...   
1      ACTA2                      Cancer-associated fibroblasts   
2    ADAMTS4                                  Matrix remodeling   
3       AKT1                                   Cancer Signaling   
4       AKT2                                   Cancer Signaling   
..       ...                                                ...   
495     XCR1                          Macrophage and DC traffic   
496     YAP1                                      Hippo pathway   
497    ZAP70                                     Effector cells   
498    ZBED2   Transcriptional regulator and competes iwht IRF1   
499     ZEB1       Transcriptional repressor in immune response   

                                           Description               ID  \
0                        atypical chemokine receptor 3  ENST00000272928   
1    Myofibroblasts acquire contractile stress fibe...  ENST00000224784   
2    ADAMTSs are a family of secreted proteinases t...  ENST00000367996   
3                        AKT serine/threonine kinase 1  ENST00000407796   
4                        AKT serine/threonine kinase 2  ENST00000392038   
..                                                 ...              ...   
495  With the recognition of cross-presenting DCs a...  ENST00000309285   
496          Yes1 associated transcriptional regulator  ENST00000282441   
497  ZAP-70 (?-associated protein of 70 kDa) is a t...  ENST00000463643   
498                  zinc finger BED-type containing 2  ENST00000317012   
499               zinc finger E-box binding homeobox 1  ENST00000320985   

        HGNC ID  Location  Unnamed: 6  Unnamed: 7  
0    HGNC:23692    2q37.3         NaN         NaN  
1           NaN       NaN         NaN         NaN  
2           NaN       NaN         NaN         NaN  
3      HGNC:391       NaN         NaN         NaN  
4      HGNC:392   19q13.2         NaN         NaN  
..          ...       ...         ...         ...  
495         NaN       NaN         NaN         NaN  
496  HGNC:16262   11q22.1         NaN         NaN  
497         NaN       NaN         NaN         NaN  
498  HGNC:20710   3q13.13         NaN         NaN  
499  HGNC:11642  10p11.22         NaN         NaN  

[500 rows x 8 columns]

In [ ]:
# แสดงเฉพาะคอลัมน์ Gene
Mela1 = Mela['ID']

# แสดงเฉพาะคอลัมน์ Gene
ENST1 = ENST['id_with_url']

# เปลี่ยนชื่อคอลัมน์ให้เหมือนกันเพื่อความสะดวกในการเปรียบเทียบ
ENST1 = ENST1.rename('ID')

# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
merged_enst = pd.merge(ENST1, Mela1, on='ID', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes_count1 = len(merged_enst)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่ตรงกันระหว่างตาราง Mela1 และ ENST1 คือ: {matched_genes_count1} ตัว")

# แสดmerged_enstงตารางที่รวมกันแล้ว (ถ้าต้องการ)
merged_enst

จำนวนยีนที่ตรงกันระหว่างตาราง Mela1 และ ENST1 คือ: 0 ตัว


Empty DataFrame
Columns: [ID]
Index: []

ไม่มีที่ match กันเลย

# NCBI => search "melanoma" => Sequence content กด "Ensembl"

การเชื่อมโยงข้อมูลจาก Ensembl Genome Browser ซึ่งเป็นฐานข้อมูลจีโนมและทรานสคริปต์ที่ครอบคลุมของสิ่งมีชีวิตหลากหลายชนิด
**Identifier ที่ใช้:**
1. Transcript ID: เช่น ENST00000288602.
2. Protein ID: เช่น ENSP00000320220.
3. Gene ID: เช่น ENSG00000157764.



In [ ]:
ncbi_enst = pd.read_csv('/content/drive/MyDrive/Spatial/ncbi_ensembl.txt', delimiter='\t')
ncbi_enst.head()

tax_id      Org_name  GeneID  CurrentID Status   Symbol  \
0    9606  Homo sapiens    1029          0   live   CDKN2A   
1    9606  Homo sapiens   51151          0   live  SLC45A2   
2    9606  Homo sapiens    1026          0   live   CDKN1A   
3    9606  Homo sapiens   64135          0   live    IFIH1   
4    9606  Homo sapiens    4157          0   live     MC1R   

                                             Aliases  \
0  ARF, CAI2, CDK4I, CDKN2, CMM2, INK4, INK4A, ML...   
1                       1A1, AIM1, MATP, OCA4, SHEP5   
2  CAP20, CDKN1, CIP1, MDA-6, P21, SDI1, WAF1, p2...   
3  AGS7, Hlcd, IDDM19, IMD95, MDA-5, MDA5, RLR-2,...   
4                                 CMM5, MSH-R, SHEP2   

                                   description  \
0         cyclin dependent kinase inhibitor 2A   
1            solute carrier family 45 member 2   
2         cyclin dependent kinase inhibitor 1A   
3  interferon induced with helicase C domain 1   
4                      melanocortin 1 receptor   

                                  other_designations map_location chromosome  \
0  cyclin-dependent kinase inhibitor 2A|CDK4 inhi...       9p21.3          9   
1  membrane-associated transporter protein|antige...       5p13.2          5   
2  cyclin-dependent kinase inhibitor 1|CDK-intera...       6p21.2          6   
3  interferon-induced helicase C domain-containin...       2q24.2          2   
4  melanocyte-stimulating hormone receptor|MC1-R|...      16q24.3         16   

  genomic_nucleotide_accession.version  \
0                         NC_000009.12   
1                         NC_000005.10   
2                         NC_000006.12   
3                         NC_000002.12   
4                         NC_000016.10   

   start_position_on_the_genomic_accession  \
0                               21967752.0   
1                               33944623.0   
2                               36676463.0   
3                              162267074.0   
4                               89918862.0   

   end_position_on_the_genomic_accession orientation  exon_count    OMIM  \
0                             21995324.0       minus        10.0  600160   
1                             33984693.0       minus         8.0  606202   
2                             36687332.0        plus         6.0  116899   
3                            162318684.0       minus        17.0  606951   
4                             89920972.0        plus         1.0  155555   

   Unnamed: 17  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN

In [ ]:
# แสดงเฉพาะคอลัมน์ Gene
Mela1 = Mela['Gene ']

# เปลี่ยนชื่อคอลัมน์ให้เหมือนกันเพื่อความสะดวกในการเปรียบเทียบ
Mela1 = Mela1.rename('Symbol')

# แสดงเฉพาะคอลัมน์ Symbol
ncbi_enst1 = ncbi_enst['Symbol']

# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
merged = pd.merge(ncbi_enst1, Mela1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes_count2 = len(merged)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่ตรงกันระหว่างตาราง Mela1 และ ncbi_enst1 คือ: {matched_genes_count2} ตัว")

# แสดmerged_enstงตารางที่รวมกันแล้ว (ถ้าต้องการ)
merged

จำนวนยีนที่ตรงกันระหว่างตาราง Mela1 และ ncbi_enst1 คือ: 316 ตัว


Symbol
0    CDKN1A
1     CXCL1
2       FAP
3     CXCL2
4     CXCL1
..      ...
311   CCR10
312   PTGS2
313    ZEB1
314     KIT
315   PDCD1

[316 rows x 1 columns]

จำนวนยีนที่ตรงกันระหว่างตาราง Mela1 และ ncbi_enst1 คือ: 316 ตัว